## TRAVAIL SUR EAU DU ROBINET 

Base EDR téléchargeable 
https://www.data.gouv.fr/fr/datasets/resultats-du-controle-sanitaire-de-leau-du-robinet/#/resources/8b59c86c-b1fd-44e0-8295-ac00f3320791

In [1]:
import pandas as pd

pd.set_option("display.max_columns", None)  # show all cols
pd.set_option("display.max_colwidth", None)  # show full width of showing cols
pd.set_option("display.expand_frame_repr", False)

In [28]:
edr_udi_plv = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\UDI_PLV_202412.txt", delimiter=","
)
edr_udi_plv.head(2)

C:\Users\EPPINGER Muriel\AppData\Local\Temp\ipykernel_10632\1610464244.py:1: DtypeWarning: Columns (0,3,4,6,11,13,23,26) have mixed types. Specify dtype option on import or set low_memory=False.
  edr_udi_plv = pd.read_csv(r'C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\UDI_PLV_202412.txt', delimiter=',')


distrcode                                                   distrlib    distrsiret cddept inseecommune                nomcommune   cdreseau              nomreseau codetypeinstallation    nomtypeinstallation inae    cdpointsurv          nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite     coord_x      coord_y
0  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  2.501018e+13    001        01002  ABERGEMENT-DE-VAREY (L')  001000369  L'ABERGEMENT-DE-VAREY                  UDI  UNITE DE DISTRIBUTION  ESO  0010000000387  L'ABERGEMENT DALIVOY   00100143433  2024-12-11     13h00           CS                                                Eau d'alimentation non-conforme aux exigences de qualité.        T   EAU DISTRIBUEE DESINFECTEE                     N                     C          O             00100003  D1                 NaN  13000807100107  250101839.0                        N                        C             1             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2025-01-06                O         NaN          NaN
1  00101028  SERA - SYNDICAT DES EAUX DE LA REGION D'AMBERIEU-EN-BUGEY  2.501018e+13    001        01002  ABERGEMENT-DE-VAREY (L')  001000369  L'ABERGEMENT-DE-VAREY                  UDI  UNITE DE DISTRIBUTION  ESO  0010000000386    L'ABERGEMENT BOURG   00100143716  2024-12-16     12h25           S3  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00100003  D1                 INF  13000807100107  250101839.0                        C                        C             1             NaN           NaN       NaN      NaN  2024-12-26  2024-12-27  2025-01-06                O  887546.901  6549138.557

In [29]:
edr_udi_plv.describe(include="all")

distrcode         distrlib    distrsiret cddept  inseecommune nomcommune    cdreseau                nomreseau codetypeinstallation    nomtypeinstallation   inae   cdpointsurv nompointsurv  referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet  plvcdorganismepayeur typevisite plvcondition     preleveur    distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite       coord_x       coord_y
count     24532.0            24532  1.558900e+04  24532       24532.0      24532     24532.0                    24532                24532                  24532  24530  2.453200e+04        24532   2.453200e+04       24532     24526        24532                                                                                                    24532     24532                       24532                 24532                 24532      24532               24421.0      24532         5733  1.254200e+04  1.558900e+04                    24532                    24532   24532.00000             0.0           0.0       0.0      0.0       24532       24532       24532            24532  1.537500e+04  1.537400e+04
unique     3453.0             3294           NaN    101       13299.0      12944     12284.0                    12197                    1                      1      4  1.994100e+04        14528   2.453200e+04         431       600           10                                                                                                     1628         6                           6                     3                     4          1                3601.0         37           25  5.500000e+01           NaN                        3                        3           NaN             NaN           NaN       NaN      NaN          44          47        1276                2           NaN           NaN
top     7500034.0  SUEZ EAU FRANCE           NaN    059       75056.0      PARIS  69000069.0  CENTRE (METROPOLE LYON)                  UDI  UNITE DE DISTRIBUTION    ESO  6.900000e+11        BOURG   9.760003e+10  2024-12-09     09h30           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             7500034.0  D1                 INF  4.105453e+13           NaN                        C                        C           NaN             NaN           NaN       NaN      NaN  2024-12-16  2024-12-17  2024-07-16                O           NaN           NaN
freq        443.0              746           NaN    900         443.0        443       139.0                      139                24532                  24532  17262  1.900000e+01         1662   1.000000e+00        1205       258        22515                                                                                                    18083     22946                       22946                 22485                 23988      24532                 443.0      21060         3490  1.818000e+03           NaN                    21683                    21412           NaN             NaN           NaN       NaN      NaN        1564        1561         424            24087           NaN           NaN
mean          NaN              NaN  4.557543e+13    NaN           NaN        NaN         NaN                      NaN                  NaN                    NaN    NaN           NaN          NaN            NaN         NaN       NaN          NaN                                                                                                      NaN       NaN                         NaN                   NaN                   NaN        NaN     

In [30]:
edr_udi_res = pd.read_csv(
    r"C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\UDI_RES_202412.txt", delimiter=","
)
edr_udi_res.head(2)

C:\Users\EPPINGER Muriel\AppData\Local\Temp\ipykernel_10632\1904612978.py:1: DtypeWarning: Columns (0,1,2,3,5,8) have mixed types. Specify dtype option on import or set low_memory=False.
  edr_udi_res = pd.read_csv(r'C:\Users\EPPINGER Muriel\Documents\D4G_13\EDR\UDI_RES_202412.txt', delimiter=',')


cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
0      1     100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       6489.0                 ASP     0     NaN    0.0               SANS OBJET                X                NaN              217.0                O
1      1     100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1042.0                BSIR    <1       <    0.0                n/(100mL)              226                NaN              217.0                O

In [31]:
edr_udi_res.describe(include="all")

cddept  referenceprel     cdanalyse refechantillonlabo datereceptionechant  cdlaboratoire   laboratoire cdtypeanalyse  analysecdorganisme analysedate insituana    cdparametre cdparametresiseeaux   rqana rssigne          rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
count   587307.0   5.873070e+05  4.547210e+05             454721              450988       454721.0  4.547210e+05        453732            454665.0      454721    587307  586538.000000              587256  587078  282087  579673.000000                   587307           587307          85.000000       3.803200e+05           587307
unique     102.0   2.453300e+04  2.478600e+04              24776                 430           52.0           NaN           866              3636.0         413         2            NaN                1011    5053       7            NaN                       35               35                NaN                NaN                2
top         51.0   5.100140e+09  5.100157e+09      H_CS24.8636.1          2024-12-10         6901.0           NaN        051D1+           1000569.0  2024-12-09         L            NaN                  PH       0       <            NaN                     µg/L              133                NaN                NaN                O
freq     27014.0   5.900000e+02  5.850000e+02                585               24442       170155.0           NaN         22441             20043.0       28165    454721            NaN               30125  111578  273444            NaN                   151567           151567                NaN                NaN           587082
mean         NaN            NaN           NaN                NaN                 NaN            NaN  3.457108e+13           NaN                 NaN         NaN       NaN    3035.665635                 NaN     NaN     NaN      22.214852                      NaN              NaN          20.882353       1.448036e+12              NaN
std          NaN            NaN           NaN                NaN                 NaN            NaN  1.675950e+13           NaN                 NaN         NaN       NaN    3961.373890                 NaN     NaN     NaN     103.529905                      NaN              NaN           6.190261       4.453987e+12              NaN
min          NaN            NaN           NaN                NaN                 NaN            NaN  1.300021e+13           NaN                 NaN         NaN       NaN    1034.000000                 NaN     NaN     NaN     -16.900000                      NaN              NaN           3.000000       9.800000e+01              NaN
25%          NaN            NaN           NaN                NaN                 NaN            NaN  2.202000e+13           NaN                 NaN         NaN       NaN    1303.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN          23.000000       2.050000e+02              NaN
50%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    1447.000000                 NaN     NaN     NaN       0.000000                      NaN              NaN          23.000000       2.170000e+02              NaN
75%          NaN            NaN           NaN                NaN                 NaN            NaN  4.105453e+13           NaN                 NaN         NaN       NaN    5441.000000                 NaN     NaN     NaN       0.400000                      NaN              NaN          23.000000       2.170000e+02              NaN
max          NaN            NaN           NaN                NaN                 NaN            NaN  7.788520e+13           NaN                 NaN         NaN       NaN   99999.000000                 NaN     NaN     NaN    5081.000000                      NaN              NaN          23.000000       2.000132e+13              NaN

In [32]:
# modification format reference prel pour comparaison avec autres bdd

edr_udi_res["referenceprel"] = edr_udi_res["referenceprel"].astype("string")
edr_udi_res["referenceprel"] = edr_udi_res["referenceprel"].apply(
    lambda x: "00" + x if len(x) == 9 else "0" + x if len(x) == 10 else x
)

In [33]:
# recherche de doublons dans la base des prélevements

doublons = edr_udi_plv.duplicated()
df_doublons_plv = edr_udi_plv[doublons]
print(df_doublons_plv)

# recherche de doublons dans la base des résultats

doublons = edr_udi_res.duplicated()
df_doublons_res = edr_udi_res[doublons]
print(df_doublons_res)

Empty DataFrame
Columns: [distrcode, distrlib, distrsiret, cddept, inseecommune, nomcommune, cdreseau, nomreseau, codetypeinstallation, nomtypeinstallation, inae, cdpointsurv, nompointsurv, referenceprel, dateprel, heureprel, finaliteprel, conclusionprel, cdtypeeau, libtypeeau, plvconformitebacterio, plvconformitechimique, plvcomplet, plvcdorganismepayeur, typevisite, plvcondition, preleveur, distrsiren, plvconformiterefbacterio, plvconformiterefchimique, cddeptsandre, dtautorisation, champcaptant, codebrgm, codebss, dmajplv, dmodifplv, dmodifins, representativite, coord_x, coord_y]
Index: []
Empty DataFrame
Columns: [cddept, referenceprel, cdanalyse, refechantillonlabo, datereceptionechant, cdlaboratoire, laboratoire, cdtypeanalyse, analysecdorganisme, analysedate, insituana, cdparametre, cdparametresiseeaux, rqana, rssigne, rsana, cdunitereferencesiseeaux, cdunitereference, cdfractionanalyse, sandrelaboratoire, representativite]
Index: []


#Cas particulier : 1 prélèvement avec 2 codes analyse différents

In [10]:
edr_udi_plv.loc[edr_udi_plv["referenceprel"] == "00100143183", :]

distrcode                              distrlib  distrsiret cddept inseecommune  nomcommune   cdreseau                   nomreseau codetypeinstallation    nomtypeinstallation inae    cdpointsurv                    nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur  distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite  coord_x  coord_y
35  00101000  CC TERRE VALSERHONE - REGIE DES EAUX         NaN    001        01033  VALSERHONE  001001033  BELLEGARDE QUARTIERS HAUTS                  UDI  UNITE DE DISTRIBUTION  ESO  0010000002222  BELLEGARDE STATION D'EPURATION   00100143183  2024-11-21     12h00           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00101000  D2                 NaN  13000807100107         NaN                        C                        C             1             NaN           NaN       NaN      NaN  2024-12-03  2024-12-04  2019-12-04                O      NaN      NaN

In [34]:
edr_udi_res.loc[
    (
        (edr_udi_res["referenceprel"] == "00100143183")
        & (edr_udi_res["cdparametresiseeaux"] == "ASP")
    ),
    :,
]

cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux rqana rssigne  rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
1611      1   00100143183  100152188.0      LSE2411-32520          2024-11-21        6901.0  4.105453e+13        0011D2           101000.0  2024-11-29         L       6489.0                 ASP     0     NaN    0.0               SANS OBJET                X                NaN              217.0                O
1612      1   00100143183  100152189.0      LSE2411-32337          2024-11-21        6901.0  4.105453e+13        0011AU           101000.0  2024-11-26         L       6489.0                 ASP     0     NaN    0.0               SANS OBJET                X                NaN              217.0                O

In [12]:
# création d'une liste unique de referenceprel

edr_plv = edr_udi_plv[["referenceprel"]].copy()
unique_edr_plv = edr_plv.drop_duplicates()
unique_edr_plv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24532 entries, 0 to 24531
Data columns (total 1 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   referenceprel  24532 non-null  object
dtypes: object(1)
memory usage: 191.8+ KB


## TRAVAIL SUR EAU DE COMMUNE

In [14]:
import duckdb
from pipelines.tasks._common import DUCKDB_FILE

con = duckdb.connect(database=DUCKDB_FILE, read_only=True)

In [15]:
query_2024 = """
SELECT * from edc_prelevements
"""

edc_plv = con.sql(query_2024)
data_edc_plv = edc_plv.df()
data_edc_plv.head(2)

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont      nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                           conclusionprel                         ugelib                        distrlib                          moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000003             01007        AMBRONAY          None                None          None   00100119766 2020-02-13     11h40  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005
1    001  001000003             01007        AMBRONAY     001001304  TTP (CLG) AMBRONAY         100 %   00100120290 2020-03-17     11h15  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.  SI REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY  SIE REGION D'AMBERIEU-EN-BUGEY                     C                     C                          C                          C          2020        2025-02-14     20230811-150005

In [46]:
# extraction d'une referenceprel dnas bdd "prélèvements"

edr_udi_plv.loc[edr_udi_plv["referenceprel"] == "00100141659", :]

distrcode     distrlib    distrsiret cddept inseecommune nomcommune   cdreseau              nomreseau codetypeinstallation    nomtypeinstallation inae    cdpointsurv           nompointsurv referenceprel    dateprel heureprel finaliteprel                                                                                 conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite     coord_x      coord_y
27  00101002  SOGEDO LYON  3.011928e+13    001        01032  BELIGNEUX  001000487  BELIGNEUX LA VALBONNE                  UDI  UNITE DE DISTRIBUTION  ESO  0010000000529  BELIGNEUX LA VALBONNE   00100141659  2024-08-12     11h56           CS  Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00100973  D1                 NaN  13000807100107  301192803.0                        C                        N             1             NaN           NaN       NaN      NaN  2024-12-03  2024-12-04  2024-10-10                O  865227.563  6530365.073

In [47]:
# extraction d'une referenceprel dnas bdd "prélèvements"

exple = """
SELECT * from edc_prelevements
WHERE referenceprel='00100141659'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

cddept   cdreseau inseecommuneprinc nomcommuneprinc cdreseauamont         nomreseauamont pourcentdebit referenceprel   dateprel heureprel                                                                                 conclusionprel                              ugelib     distrlib                             moalib plvconformitebacterio plvconformitechimique plvconformitereferencebact plvconformitereferencechim  de_partition de_ingestion_date de_dataset_datetime
0    001  001000487             01032       BELIGNEUX          None                   None          None   00100141659 2024-08-12     11h56  Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité.  CC COTIERE A MONTLUEL (3CM) SOGEDO  SOGEDO LYON  CC DE LA CÔTIÈRE À MONTLUEL (3CM)                     C                     C                          C                          N          2024        2025-02-14     20250203-075802
1    001  001000488             01032       BELIGNEUX     001000487  BELIGNEUX LA VALBONNE           0 %   00100141659 2024-08-12     11h56  Eau d'alimentation conforme aux limites de qualité et non conforme aux références de qualité.  CC COTIERE A MONTLUEL (3CM) SOGEDO  SOGEDO LYON  CC DE LA CÔTIÈRE À MONTLUEL (3CM)                     C                     C                          C                          N          2024        2025-02-14     20250203-075802

La referenceprel est identique entre EDR et EDC et retourne les mêmes données dans la base "prélèvements"

In [48]:
# extraction d'un prélèvement dans bdd resultats

edr_udi_res.loc[edr_udi_res["referenceprel"] == "00100141659", :]

cddept referenceprel    cdanalyse refechantillonlabo datereceptionechant cdlaboratoire   laboratoire cdtypeanalyse analysecdorganisme analysedate insituana  cdparametre cdparametresiseeaux  rqana rssigne   rsana cdunitereferencesiseeaux cdunitereference  cdfractionanalyse  sandrelaboratoire representativite
0       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       6489.0                 ASP      0     NaN    0.00               SANS OBJET                X                NaN              217.0                O
1       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1042.0                BSIR     <1       <    0.00                n/(100mL)              226                NaN              217.0                O
2       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1303.0               CDT25    554     NaN  554.00                    µS/cm              147                NaN              217.0                O
3       1   00100141659          NaN                NaN                 NaN           NaN           NaN           NaN                NaN         NaN         T       1398.0              CL2LIB   0,12     NaN    0.12                mg(Cl2)/L              165                NaN                NaN                O
4       1   00100141659          NaN                NaN                 NaN           NaN           NaN           NaN                NaN         NaN         T       1399.0              CL2TOT   0,16     NaN    0.16                mg(Cl2)/L              165                NaN                NaN                O
5       1   00100141659          NaN                NaN                 NaN           NaN           NaN           NaN                NaN         NaN         T       1754.0                CLO2   N.M.       N     NaN                     mg/L              162                NaN                NaN                O
6       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1309.0               COULF     <5       <    0.00                 mg(Pt)/L              178                NaN              217.0                O
7       1   00100141659          NaN                NaN                 NaN           NaN           NaN           NaN                NaN         NaN         T       5900.0               COULQ      0     NaN    0.00               SANS OBJET                X                NaN                NaN                O
8       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1447.0                 CTF     <1       <    0.00                n/(100mL)              226                NaN              217.0                O
9       1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       1449.0               ECOLI     <1       <    0.00                n/(100mL)              226                NaN              217.0                O
10      1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       5440.0             GT22_68     <1       <    0.00                     n/mL              222                NaN              217.0                O
11      1   00100141659  100150585.0      LSE2408-35603          2024-08-12        6901.0  4.105453e+13        0011DN           100973.0  2024-11-27         L       5441.0             GT36_44     <1       <    0.00                     n/mL              222   

In [49]:
exple_res = """
SELECT * from edc_resultats
WHERE referenceprel='00100141659'
"""

edc_res_exple = con.sql(exple_res)
data_edc_res_exple = edc_res_exple.df()
data_edc_res_exple

cddept referenceprel cdparametresiseeaux  cdparametre                      libmajparametre                      libminparametre libwebparametre qualitparam insituana                     rqana cdunitereferencesiseeaux cdunitereference     limitequal                refqual  valtraduite    casparam referenceanl  de_partition de_ingestion_date de_dataset_datetime
0     001   00100141659                 ASP         6489                  ASPECT (QUALITATIF)                  Aspect (qualitatif)            None           O         L             Aspect normal               SANS OBJET                X           None                   None         0.00        None  00100150585          2024        2025-02-14     20250203-075802
1     001   00100141659                BSIR         1042  BACT. ET SPORES SULFITO-RÉDU./100ML  Bact. et spores sulfito-rédu./100ml            None           N         L                        <1                n/(100mL)              226           None          <=0 n/(100mL)         0.00        None  00100150585          2024        2025-02-14     20250203-075802
2     001   00100141659               CDT25         1303                  CONDUCTIVITÉ À 25°C                  Conductivité à 25°C            None           N         L                       554                    µS/cm              147           None  >=200 et <=1100 µS/cm       554.00        None  00100150585          2024        2025-02-14     20250203-075802
3     001   00100141659              CL2LIB         1398                         CHLORE LIBRE                         Chlore libre            None           N         T                      0,12                mg(Cl2)/L              165           None                   None         0.12        None         None          2024        2025-02-14     20250203-075802
4     001   00100141659              CL2TOT         1399                         CHLORE TOTAL                         Chlore total            None           N         T                      0,16                mg(Cl2)/L              165           None                   None         0.16        None         None          2024        2025-02-14     20250203-075802
5     001   00100141659                CLO2         1754          BIOXYDE DE CHLORE MG/L CLO2          Bioxyde de chlore mg/L ClO2            None           N         T                      N.M.                     mg/L              162           None                   None          NaN  10049-04-4         None          2024        2025-02-14     20250203-075802
6     001   00100141659               COULF         1309                           COLORATION                           Coloration            None           N         L                        <5                 mg(Pt)/L              178           None          <=15 mg(Pt)/L         0.00        None  00100150585          2024        2025-02-14     20250203-075802
7     001   00100141659               COULQ         5900                 COULEUR (QUALITATIF)                 Couleur (qualitatif)            None           O         T  Aucun changement anormal               SANS OBJET                X           None                   None         0.00        None         None          2024        2025-02-14     20250203-075802
8     001   00100141659                 CTF         1447       BACTÉRIES COLIFORMES /100ML-MS       Bactéries coliformes /100ml-MS            None           N         L                        <1                n/(100mL)              226           None          <=0 n/(100mL)         0.00        None  00100150585          2024        2025-02-14     20250203-075802
9     001   00100141659               ECOLI         1449         ESCHERICHIA COLI /100ML - MF         Escherichia coli /100ml - MF            None           N         L                        <1                n/(100mL)              226  <=0 n/(100mL)                   None         0.00        None  00100150585          2024        2025-02-14     2025

In [35]:
# recherche liste unique referenceprel

edc_plv = data_edc_plv[["referenceprel"]].copy()
unique_edc_plv = edc_plv.drop_duplicates()
unique_edc_plv.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1436114 entries, 0 to 2083344
Data columns (total 1 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   referenceprel  1436114 non-null  object
dtypes: object(1)
memory usage: 21.9+ MB


In [36]:
# recherche des prélèvements présents dans EDR et absents dans EDC

resultat = unique_edr_plv[
    ~unique_edr_plv["referenceprel"].isin(unique_edc_plv["referenceprel"])
]
print(resultat)

      referenceprel
444     00200202245
445     00200202246
446     00200202247
881     00500133728
889     00500134152
...             ...
24527   97600032413
24528   97600032302
24529   97600032298
24530   97600032326
24531   97600032321

[8588 rows x 1 columns]


In [38]:
# extraction des 8588 prélèvements absents de EDC

res = resultat.merge(
    edr_udi_plv, left_on="referenceprel", right_on="referenceprel", how="inner"
)
res.describe(include="all")

referenceprel  distrcode         distrlib    distrsiret  cddept  inseecommune nomcommune    cdreseau                nomreseau codetypeinstallation    nomtypeinstallation  inae   cdpointsurv  nompointsurv    dateprel heureprel finaliteprel                                                                                           conclusionprel cdtypeeau                  libtypeeau plvconformitebacterio plvconformitechimique plvcomplet  plvcdorganismepayeur typevisite plvcondition     preleveur    distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite       coord_x       coord_y
count    8.588000e+03     8588.0             8588  6.124000e+03  8588.0        8588.0       8588      8588.0                     8588                 8588                   8588  8587  8.588000e+03          8588        8588      8587         8588                                                                                                     8588      8588                        8588                  8588                  8588       8588                8582.0       8588         1942  4.113000e+03  6.124000e+03                     8588                     8588   8588.000000             0.0           0.0       0.0      0.0        8588        8588        8588             8588  5.213000e+03  5.212000e+03
unique   8.588000e+03     1124.0             1052           NaN    85.0        4013.0       3963      3625.0                     3609                    1                      1     4  6.854000e+03          5688         370       575            6                                                                                                      453         6                           6                     3                     4          1                1195.0         22           18  4.300000e+01           NaN                        3                        3           NaN             NaN           NaN       NaN      NaN          43          46         674                2           NaN           NaN
top      9.760003e+10  7500034.0  SUEZ EAU FRANCE           NaN    78.0       75056.0      PARIS  69000069.0  CENTRE (METROPOLE LYON)                  UDI  UNITE DE DISTRIBUTION   ESO  6.900000e+11  CENTRE BOURG  2024-12-02     10h30           CS  Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T   EAU DISTRIBUEE DESINFECTEE                     C                     C          O             7500034.0  D1                 INF  4.105453e+13           NaN                        C                        C           NaN             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2024-07-16                O           NaN           NaN
freq     1.000000e+00      443.0              458           NaN   486.0         443.0        443       139.0                      139                 8588                   8588  5353  1.900000e+01           500         463        80         7649                                                                                                     6393      8225                        8225                  7748                  8508       8588                 443.0       7148         1073  1.818000e+03           NaN                     7491                     7332           NaN             NaN           NaN       NaN      NaN         534         534         419             8144           NaN           NaN
mean              NaN        NaN              NaN  4.695672e+13     NaN           NaN        NaN         NaN                      NaN                  NaN                    NaN   NaN           NaN           NaN         NaN       NaN          NaN                                                                                                      NaN       NaN                         NaN                   NaN                   NaN        Na

In [51]:
res.head(10)

referenceprel distrcode                           distrlib    distrsiret cddept inseecommune             nomcommune   cdreseau                     nomreseau codetypeinstallation    nomtypeinstallation inae    cdpointsurv                    nompointsurv    dateprel heureprel finaliteprel                                                                                                                                                                                                                                                                                                                                                            conclusionprel cdtypeeau                        libtypeeau plvconformitebacterio plvconformitechimique plvcomplet plvcdorganismepayeur typevisite plvcondition       preleveur   distrsiren plvconformiterefbacterio plvconformiterefchimique  cddeptsandre  dtautorisation  champcaptant  codebrgm  codebss     dmajplv   dmodifplv   dmodifins representativite     coord_x      coord_y
0   00200202245  00200919                ARS HAUTS-DE-FRANCE  1.300080e+13    002        02722               SOISSONS  002001532               UDI DE SOISSONS                  UDI  UNITE DE DISTRIBUTION  ESO  0020000002104       SECTEUR VILLENEUVE ANCIEN  2024-12-12     08h44           CS                                                                                                                                                                                                                                                                   Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T         EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00200919  D1                 NaN  13000797400079  130007974.0                        C                        C             2             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2024-12-03                O         NaN          NaN
1   00200202246  00200919                ARS HAUTS-DE-FRANCE  1.300080e+13    002        02722               SOISSONS  002001532               UDI DE SOISSONS                  UDI  UNITE DE DISTRIBUTION  ESO  0020000002103            SECTEUR PERIPHERIQUE  2024-12-12     09h14           CS                                                                                                                                                                                                                                                                   Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T         EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00200919  D1                 NaN  13000797400079  130007974.0                        C                        C             2             NaN           NaN       NaN      NaN  2024-12-17  2024-12-18  2024-12-03                O         NaN          NaN
2   00200202247  00200919                ARS HAUTS-DE-FRANCE  1.300080e+13    002        02722               SOISSONS  002001532               UDI DE SOISSONS                  UDI  UNITE DE DISTRIBUTION  ESO  0020000002101                  SECTEUR CENTRE  2024-12-12     08h58           CS                                                                                                                                                                                                                                                                   Eau d'alimentation conforme aux exigences de qualité en vigueur pour l'ensemble des paramètres mesurés.        T         EAU DISTRIBUEE DESINFECTEE                     C                     C          O             00200919  D1                 NaN  13000797400079  130007974.0                        C                        C             2             NaN           NaN       NaN      NaN  2024-12-17  202

In [52]:
exple = """
SELECT * from edc_prelevements
WHERE inseecommuneprinc='05053' and dateprel='2024-11-21'
"""

edc_plv_exple = con.sql(exple)
data_edc_plv_exple = edc_plv_exple.df()
data_edc_plv_exple

Empty DataFrame
Columns: [cddept, cdreseau, inseecommuneprinc, nomcommuneprinc, cdreseauamont, nomreseauamont, pourcentdebit, referenceprel, dateprel, heureprel, conclusionprel, ugelib, distrlib, moalib, plvconformitebacterio, plvconformitechimique, plvconformitereferencebact, plvconformitereferencechim, de_partition, de_ingestion_date, de_dataset_datetime]
Index: []

In [45]:
res.groupby("cddept")["referenceprel"].count()

cddept
64     458
65     202
66     239
67     289
68     282
      ... 
060      1
061      2
062      1
063      1
064     42
Name: referenceprel, Length: 85, dtype: int64

CONCLUSION : 

La base EDR UDI contient 24532 prélèvements
8588 de ces prélèvements sont absents de la base EDC (soit 35%). Des vérifications on été faites  sur la date de prélèvement et le département pour qq prélèvements pour pallier à une anomalie sur la référence. Aucune correspondande n'a été détectée.
Les prélèvements absents de EDC sont répartis sur 85 départements.

Réfléchir sur l'intérêt d'étendre ces résultats sur la totalité des années 
Réfléchir sur l'intérêt d'intégrer ces informations dans la base EDC